modules/images/webp-uploads/load.php

```diff
@@ -573,3 +573,106 @@ function webp_uploads_update_featured_image( $html, $post_id, $attachment_id ) {
 	return webp_uploads_img_tag_update_mime_type( $html, 'post_thumbnail_html', $attachment_id );
 }
 add_filter( 'post_thumbnail_html', 'webp_uploads_update_featured_image', 10, 3 );
+
+/**
+ * Adds a fallback mechanism to replace webp images with jpeg alternatives on older browsers.
+ *
+ * @since n.e.x.t
+ */
+function webp_uploads_wepb_fallback() {
+	// Get mime type transofrms for the site.
+	$transforms = webp_uploads_get_upload_image_mime_transforms();
+	if ( ! is_array( $transforms ) ) {
+		return;
+	}
+
+	// We need to add fallback only if jpeg alternatives for the webp images are enabled for the server.
+	$jpegs = in_array( 'image/jpeg', $transforms['image/jpeg'], true ) && in_array( 'image/webp', $transforms['image/jpeg'], true );
+	$webp  = in_array( 'image/jpeg', $transforms['image/webp'], true );
+	if ( ! $jpegs && ! $webp ) {
+		return;
+	}
+
+	?>
+	<script>
+		( function() {
+			window.fallbackWebpImages = function( media ) {
+				for ( var i = 0; i < media.length; i++ ) {
+					try {
+						var ext = media[i].media_details.sources['image/jpeg'].file.match( /\.\w+$/i );
+						if ( ! ext || ! ext[0] ) {
+							continue;
+						}
+
+						var images = document.querySelectorAll( 'img.wp-image-' + media[i].id );
+						for ( var j = 0; j < images.length; j++ ) {
+							images[j].src = images[j].src.replace( /\.webp$/i, ext[0] );
+							var srcset = images[j].getAttribute( 'srcset' );
+							if ( srcset ) {
+								images[j].setAttribute( 'srcset', srcset.replace( /\.webp(\s)/ig, ext[0] + '$1' ) );
+							}
+						}
+					} catch ( e ) {
+					}
+				}
+			};
+
+			var loadMediaDetails = function( nodes ) {
+				var ids = [];
+				for ( var i = 0; i < nodes.length; i++ ) {
+					if ( nodes[i].nodeName !== "IMG" || ! nodes[i].src.match( /\.webp$/i ) ) {
+						continue;
+					}
+
+					var attachment = nodes[i].className.match( /wp-image-(\d+)/i );
+					if ( attachment && attachment[1] && ids.indexOf( attachment[1] ) === -1 ) {
+						ids.push( attachment[1] );
+					}
+				}
+
+				if ( ! ids.length ) {
+					return;
+				}
+
+				for ( var page = 0, pages = Math.ceil( ids.length / 100 ); page < pages; page++ ) {
+					var pageIds = [];
+					for ( var i = 0; i < 100 && i + page * 100 < ids.length; i++ ) {
```

Comment:
Not sure what's the reason behind the (i+page * 100), and the nested loop. Just curious about the reason.

Reply:
100 is the max value for `per_page`.